client/core/core.go

```diff
@@ -5203,6 +5241,82 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 		tracker.changeLocked = true
 	}
 
+	if form.Queue {
+		tempID, err := order.IDFromBytes(encode.RandomBytes(order.OrderIDSize))
+		if err != nil {
+			return nil, fmt.Errorf("unable to generate temporary order ID: %w", err)
+		}
+		tracker.metaData.TempID = tempID
+		corder := tracker.coreOrder()
+
+		// Account for queued order until a server response is received.
+		dc.queuedTradesMtx.Lock()
+		dc.queuedTrades[tempID] = tracker
+		dc.queuedTradesMtx.Unlock()
+
+		defer func() {
+			dc.queuedTradesMtx.Lock()
+			delete(dc.queuedTrades, tempID)
+			dc.queuedTradesMtx.Unlock()
+		}()
+
+		// Return and send silent note for queued order.
+		tempOrderChan <- corder
+		close(tempOrderChan)
+		topic := TopicOrderQueued
+		c.notify(newOrderNote(topic, "", "", db.Data, corder))
+	}
+
+	// failedOrderError avail the provided error for queued order error
+	// notification and returns the error.
+	failedOrderError := func(err error) error {
+		if form.Queue {
+			// We only want to send order error notifications if the order was processed
+			// async.
+			topic := TopicQueueOrderFailure
+			subject, details := c.formatDetails(topic, tracker.metaData.TempID.Token(), err)
+			c.notify(newOrderNote(topic, subject, details, db.ErrorLevel, tracker.coreOrder()))
+		}
+		return err
+	}
+
+	// Send and get the result.
+	result := new(msgjson.OrderResult)
+	err = dc.signAndRequest(msgOrder, route, result, fundingTxWait+DefaultResponseTimeout)
+	if err != nil {
+		// At this point there is a possibility that the server got the request
+		// and created the trade order, but we lost the connection before
+		// receiving the response with the trade's order ID. Any preimage
+		// request will be unrecognized. This order is ABANDONED.
+		return nil, failedOrderError(fmt.Errorf("new order request with DEX server %v market %v failed: %w", dc.acct.host, mktID, err))
+	}
+
+	// If we encounter an error, perform some basic logging.
+	logAbandon := func(err error) {
+		c.log.Errorf("Abandoning order. preimage: %x, server time: %d: %v", preImg[:], result.ServerTime, err)
+	}
+
+	err = validateOrderResponse(dc, result, ord, msgOrder) // stamps the order, giving it a valid ID
```

Comment:
An issue with having the `order.Order` accessible to the app before this point via the `queuedTrades` map is that the `order.Order` is not ready, and only in `validateOrderResponse` do we stamp the order and compute+cache the order ID.  You have conditions in `coreOrder()`, but it's still a data race with the modifications here.

In the `trackedTrade`, it the `order.Order` is intended to be immutable, but it's is no longer.

If the `queuedTrades` map were a `map[order.OrderID]*db.OrderMetaData`, we wouldn't have that issue.